### ISSUE TITLE
Run scijava formatter before release

### ISSUE THREAD

**Issue by hinerm** [scijava/scijava#104]
Currently, applying [scijava coding style](https://github.com/scijava/scijava-coding-style#usage) creates a large number of code changes. We're holding off on applying it right now until everything has been merged back to `main`, to avoid the complication of merging formatting code. But everything should be updated and styled consistently before release.

We should verify that after running the maven formatter that formatting in IntelliJ and Eclipse do not cause additional changes.

**Comment by gselzer** [comment]
@hinerm @ctrueden what do you think about validating proper formatting in the build? It looks pretty easy to just activate enforcement with the existing plugin we use - the only annoyance is that formatting can only run during the `validate` goal, which happens before testing...

The other option would be to enable formatting during the CI build, but that could also be difficult...

**Comment by hinerm** [comment]
>what do you think about validating proper formatting in the build?

@gselzer I am a fan of just validating and failing the build when there are formatting differences

**Comment by ctrueden** [comment]
My personal opinion is that formatting should be manual, not part of the build. But I am OK with others overriding me on this point, if you want the build to fail when applying a formatter would result in any changes.

Another option we discussed in the past was to make `release-version.sh` do a formatting pass, which would be OK with me.

Whatever autoformat we use needs to support suppressing its behavior around blocks of code, for situations where humans know better.

**Comment by ctrueden** [comment]
The main thing is: this needs to be done at a time when there are zero pending topic branches, to avoid horrible future conflicts.

**Comment by hinerm** [comment]
Re: automation, a separate GitHub action that checks formatting would be nice. It doesn't have to fail the main build/deploy cycle but could still notify us if formatting needs to be run.

**Comment by gselzer** [comment]
I'm just going to mark this as "blocked by release" - because we probably shouldn't do it until we're ready to start releasing things.

**Comment by gselzer** [comment]
@ctrueden @hinerm unfortunately there is (currently) no way to return from a GitHub Actions step with a warning - the best we can do is something like what I've done in scijava/incubator#114

I'm really not opposed to that change, because you can still see that the build passed but it still provides the visual cues to prevent you from merging right away. Thoughts?

**Comment by hinerm** [comment]
> @ctrueden @hinerm unfortunately there is (currently) no way to return from a GitHub Actions step with a warning - the best we can do is something like what I've done in [scijava/incubator#114](https://github.com/scijava/incubator/pull/114)
> 
> I'm really not opposed to that change, because you can still see that the build passed but it still provides the visual cues to prevent you from merging right away. Thoughts?

I think a build that fails is totally fine. @gselzer are you worried this is too aggressive? That we don't want to block PRs that break formatting from being merged? If so I think it would be fine to move the config to `build-main` instead of `build-pr`

@ctrueden do you prefer the fail-fast approach on `build-pr`, or the "accept anything, but notify" on `build-main`?

**Comment by gselzer** [comment]
> @ctrueden do you prefer the fail-fast approach on `build-pr`, or the "accept anything, but notify" on `build-main`?

It seems wrong to me that we would "fail" the build on `main` but pass the build on PRs - I'd prefer the opposite.

**Comment by hinerm** [comment]
> Does the `formatter-maven-plugin` connect to https://github.com/scijava/scijava-coding-style in any way? @ctrueden we still want to use those styles right?

As identified by @gselzer, [yes it does](https://github.com/scijava/pom-scijava-base/blob/e8c15463f2c83e4a2dc511422e1322e6bba0621d/pom.xml#L869)

**Comment by ctrueden** [comment]
I have no strong opinion on whether we use the SciJava vs ImgLib2 style. I guess I lean sllightly toward ImgLib2, since it would make external collaborators happier. Ideally we would pick a style that minimizes the clashes between command-line formatting vs formatting in IDEs, especially IntelliJ. Otherwise we will keep having "formatting fights" between autoformatters... or need to turn off the IDE's autoformatter in favor of just letting CI builds fail and fixing by hand each time. The ideal solution would be for the two autoformatting mechanisms to be fully aligned.

**Comment by gselzer** [comment]
> Ideally we would pick a style that minimizes the clashes between command-line formatting vs formatting in IDEs, especially IntelliJ.

If you set up IntelliJ to use the SciJava Coding style, it will get you very close. If you import that as an Eclipse XML profile, and then set the line wrapping to 80 (mine defaulted to 120), I get very few issues...
